### ISSUE TITLE
Allman Style: No error for wrong brace position

### ISSUE THREAD

**Issue by swernerx** [eslint/eslint#5549]
**What version of ESLint are you using?**
v2.3.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
Espree

**What did you do? Please include the actual source code causing the issue.**

**Correct Code:**

```js
function sayHello()
{
  console.log("hello")
}

var object =
{
  key : 123,
  foo : "bar"
}

var array =
[
  "hello",
  "world"
]

class MyClass
{
  constructor(x, y)
  {
    this.x = x
    this.y = y
  }
}
```

**Wrong Code:**

```js
function sayHello() {
  console.log("hello")
}

var object = {
  key : 123,
  foo : "bar"
}

var array = [
  "hello",
  "world"
]

class MyClass {
  constructor(x, y) {
    this.x = x
    this.y = y
  }
}
```

**Configuration:**

```yml
env:
  browser: true
  es6: true
  node: true
  worker: true
  serviceworker: true

parserOptions:
  ecmaVersion: 6
  sourceType: module

rules:
  indent:
    - error
    - 2
    - SwitchCase: 1
      VariableDeclarator: 0

  linebreak-style:
    - error
    - unix

  quotes:
    - error
    - double
    - avoid-escape

  semi:
    - error
    - never

  brace-style:
    - error
    - allman
    - allowSingleLine: true
```

**What did you expect to happen?**

That the wrong code block (the second one) is actually reporting a few errors on wrong "brace-style".

**What actually happened? Please include the actual, raw output from ESLint.**

Both codes are passing the eslint run without any errors. It seems like the allman brace style is completely ignoring what's wrong.

**Comment by swernerx** [comment]
One part of the issue seems to be the result of the detailed configuration of `allowSingleLine`. If this setting is not defined it reports these issues:

```
⚡ eslint brace-*

/Users/Sebastian/eslinttest/brace-wrong.js
   1:1   error  Opening curly brace appears on the same line as controlling statement  brace-style
  16:14  error  Opening curly brace appears on the same line as controlling statement  brace-style

✖ 2 problems (2 errors, 0 warnings)
```

That's still ignoring array, objects and es6 classes.

**Comment by BYK** [comment]
Looks like a duplicate of #5197. Feel free to reopen if you disagree.
